### DISCUSSION TITLE
Converting CRA to NX

### DISCUSSION THREAD

**Discussion by borgdrone7** [nrwl/nx#9927] [General]
Hi,
I converted my CRA (which is JS not TS) with cra-to-nx and the process completed successfully without a single error. However when I try to serve my nx app I get this exception:
```
(node:108358) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'indexOf' of undefined
    at my-cra-app/apps/my-nx-app/craco.config.js:27:33
    at Array.find (<anonymous>)
    at my-cra-app/apps/my-nx-app/craco.config.js:26:41
    at Array.forEach (<anonymous>)
    at configure (my-cra-app/apps/my-nx-app/craco.config.js:24:29)
    at giveTotalControl (my-cra-app/node_modules/@craco/craco/lib/features/webpack/merge-webpack-config.js:57:25)
    at mergeWebpackConfig (my-cra-app/node_modules/@craco/craco/lib/features/webpack/merge-webpack-config.js:106:38)
    at overrideWebpackDev (my-cra-app/node_modules/@craco/craco/lib/features/webpack/override.js:11:36)
    at my-cra-app/node_modules/@craco/craco/scripts/start.js:27:5
(Use `node --trace-warnings ...` to show where the warning was created)
(node:108358) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:108358) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Done in 1.29s.
``` 

It happens in this part of the code:
`
            const babelLoader = r.oneOf.find(
              (rr) => rr.loader.indexOf('babel-loader') !== -1
            );
`
Anyone had similar issue? Any idea on how to resolve this?

Also I wanted to ask if I wanted to completely get rid of CRA, could I simply take all my source files and move them in newly created NX app inside app folder. Would that work? I really don't need anything from CRA itself, I just want my code to work in NX so all build/test/storybook, etc. configs can be start a new using NX and old stuff which CRA generated I don't need at all. 
Please note that current CRA app is JS and I will be adding new features to it in TS. So basically NX app needs to run my old JS code but also any new React code I write in TS should work together with the old JS code.

Thanks

**Comment by borgdrone7** [comment]
So I fixed above error by adding rr.loader && rr.loader.indexOf... (since I saw that not all modules have loader and babel ones have). However then I got bunch of module not find errors when trying to serve. So then I tried running yarn inside NX workspace but modules were already up to date. So I run yarn inside the app and it installed some modules. Now nx run myapp:serve gives me this:
`
my-cra-app/apps/my-nx-app/node_modules/react-scripts/scripts/start.js:19
  throw err;
  ^

TypeError: Cannot add property strict, object is not extensible
    at verifyTypeScriptSetup (my-cra-app/apps/my-nx-app/node_modules/react-scripts/scripts/utils/verifyTypeScriptSetup.js:231:45)
    at Object.<anonymous> (my-cra-app/apps/my-nx-app/node_modules/react-scripts/scripts/start.js:31:1)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:101:18)
    at start (my-cra-app/node_modules/@craco/craco/lib/cra.js:218:5)
    at my-cra-app/node_modules/@craco/craco/scripts/start.js:30:5
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
`

At this point I actually don't want to have anything to do with CRA, esspecially with craco. So can you please suggest to me how can I just use the old code and use everything related to build from NX and completely remove all CRA build stuff? According to NX docs for manual migration I should be able to move the code inside app, move over packages from package.json (except build stuff) and that should work. Sounds too easy to be true to be honest. Will I be able to mix JS and TS if I do it this way?

Thanks

**Comment by welingtonms** [comment]
+1
